### PR TITLE
adding lesson 3 files

### PR DESCRIPTION
**Pull request by Reem0330**
thank you for your time to review my work, still kind of struggling working with tuples

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@Reem0330 can you let us know your challenges with tuples? be happy to clarify 🙂

File: students/ReemAlqaysi/Lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    print(mylist)
+    #Add another fruit to the beginning of the list using insert() and display the list.
+    mylist.insert(0,'Mango')
+    #Display all the fruits that begin with “P”, using a for loop.
+    print("Display all the fruits that begin with “P”, using a for loop.")
+    for name in mylist:
+        if name[0].upper() == "P":
+            print(name)
+        else:
+            continue
```
Comment (natasha-aleksandrova) [inline students/ReemAlqaysi/Lesson03/list_lab.py]: no need for this continue here - it will continue the loop on its own

**Comment by Reem0330** [comment]
> @Reem0330 can you let us know your challenges with tuples? be happy to clarify 🙂

still confused on when I can use it and how to append to it
especially  if there is no specific instruction on what to use in the code list vs tuple.
to add values last time += didn't work for me, so I either change the list to list , then tuple again or as I did in one of the exercises I move it to a variable type list then move it back to the tuple in a dict.. not sure if what I'm doing is a best practice or not.

thank you for quick response
Reem

**Comment by natasha-aleksandrova** [comment]
ah yes! great questions.

Compared to a list, tuple is immutable data structure which you means you *cannot* change it.
Once you define something as a tuple its size cannot be changed. You use a tuple to ensure that it cannot be changed and to in a way "communicate" that it cannot be changed to reduce bugs and to  avoid from people changing it (if someone is using your library).

For example, in the mailroom assignment there is a good place for a tuple.

```
donors = [('Bob', [100, 200])]
```
above is a hard definition of your donor data structure, you can add another donor, but you are ensuring donor is a two item tuple with a name and a list of donations. This will prevent anyone that wants to make changes to the code from breaking that contract.


So going forward when you need to choose a data structure - ask yourself "is this data expandable? can I append to it? or will it never change?"

hope this helps!

**Comment by Reem0330** [comment]
Oh, that really makes sense now, I appreciate your time explaining the difference.. thank you so much 

Regards
Reem

**Comment by natasha-aleksandrova** [comment]
glad to hear!

**Pull request merged**